jobsapp/api/views/users.py

```diff
@@ -8,11 +8,11 @@
 class UsersViewsList(ListAPIView):
     serializer_class = UserSerializer
     queryset = User.objects.all()
-    permission_classes = [AllowAny]
+    permission_classes = [IsAuthenticatedOrReadOnly]
 
 
 class UsersViewDetails(CreateAPIView, RetrieveUpdateDestroyAPIView):
     serializer_class = UserSerializer
     queryset = User.objects.all()
     print(queryset)
-    permission_classes = [AllowAny]
+    permission_classes = [IsAuthenticatedOrReadOnly]
```

Comment:
We should add a new permission to be used here. For example `IsSelfOrReadOnly`

```python
class IsSelfOrReadOnly(BasePermission):
    # TODO: Improve message
    message = "Unauthorized"

    def has_object_permission(self, request: Request, view: View, obj: Any) -> bool:
        """
        Object-level permission to only allow user model manipulation
        """
        return bool(request.method in permissions.SAFE_METHODS or request.user.id == obj.id)
 ```

Reply:
Should also admins create users? Any requirements for this @francescarpi ?

Reply:
Anyway, we think that this should be in another PR, so for now we restrict only for itself and in future we maybe should change this.

Reply:
```python
from rest_framework.permissions import IsAdminUser
(...)
permission_classes=[IsSelfOrReadOnly|IsAdminUser]
```

Reply:
mmm, I agree to add this feature (admin permissions) in another PR if it's necessary. I'm not sure if it is a requirement, right now.